mkdocs.yml

```diff
@@ -40,8 +40,33 @@ theme:
 plugins:
   - awesome-pages
   - search
+  - macros:
+      module_name: docs/scripts/doc_macros
+  - mkdocstrings:
+      default_handler: python
+      handlers:
+        python:
+          options:
+            docstring_style: google
+            show_root_heading: true
+            show_if_no_docstring: true
+            inherited_members: true
+            members_order: source
+            separate_signature: true
+            unwrap_annotated: true
+            filters:
+              - "!^_"
+            docstring_section_style: list
+            merge_init_into_class: true
+            signature_crossrefs: true
+            show_symbol_type_heading: true
+            show_symbol_type_toc: true
   - git-revision-date-localized:
       type: timeago
+  - gen-files:
+      scripts:
+        - docs/scripts/gen_ref_pages.py
+  - section-index
```

Comment:
This is causing the section get hidden in the navigation section itself, unless there is going to be `index.md` in each of the folder with some over about the section, I believe will cause bit of confusion to readers. 

![image](https://github.com/HyphaApp/hypha/assets/236356/4f8bb4be-e91f-4a65-979b-8ba609a397f8)

Reply:
Without `section-index` plugin. 

![Screenshot 2024-01-25 at 6  19 21@2x](https://github.com/HyphaApp/hypha/assets/236356/00d59880-4379-4b1b-a846-167070961266)

Reply:
theme feature `navigation.indexes` (already enabled) probably serves the same purpose, but doesn't link the section header, if the `index.md` is missing. 

https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages

Reply:
Yeah you're totally right, I didn't even notice `section-index` was duplicating the `navigation.indexes` functionality. I tossed `section-index` and all seems to work as expected from what I've seen!